src/modules/firewall/src/lib/Firewall.h

```diff
@@ -19,169 +24,135 @@ class FirewallLog
-    void SetRawOptions(std::string rawOptions);
-
-    int GetRuleNum();
-    std::string GetTarget();
-    std::string GetProtocol();
-    std::string GetSource();
-    std::string GetDestination();
-    std::string GetSourcePort();
-    std::string GetDestinationPort();
-    std::string GetInInterface();
-    std::string GetOutInterface();
-    std::string GetRawOptions();
-
-private:
-    int m_ruleNum;
-    std::string m_policy;
-    std::string m_target;
-    std::string m_protocol;
-    std::string m_source;
-    std::string m_destination;
-    std::string m_sourcePort;
-    std::string m_destinationPort;
-    std::string m_inInterface;
-    std::string m_outInterface;
-    std::string m_rawOptions;
+    std::string Hash(const std::string str);
+    int Execute(const std::string command, std::string& result);
+    int Execute(const std::string command);
 };
 
-enum FirewallStateCode
+class GenericUtility
 {
-    firewallStateCodeUnknown = 0,
-    firewallStateCodeEnabled,
-    firewallStateCodeDisabled
-};
+public:
+    enum class State
+    {
+        Unknown = 0,
+        Enabled,
+        Disabled
+    };
 
-enum UtilityStatusCode
-{
-    utilityStatusCodeUnknown = 0,
-    utilityStatusCodeInstalled,
-    utilityStatusCodeNotInstalled
-};
+    virtual ~GenericUtility() = default;
 
-enum RuleToken
-{
-    ruleTokenNumIndex = 1,
-    ruleTokenPacketsIndex,
-    ruleTokenBytesIndex,
-    ruleTokenTargetIndex,
-    ruleTokenProtocolIndex,
-    ruleTokenOptIndex,
-    ruleTokenInIndex,
-    ruleTokenOutIndex,
-    ruleTokenSourceIndex,
-    ruleTokenDestinationIndex,
-    ruleTokenRawOptionsIndex
+    virtual State Detect() const = 0;
+    virtual std::string Hash() const = 0;
 };
 
-class Chain
+class Iptables : public GenericUtility
 {
 public:
-    Chain();
-    Chain(std::string chainName);
-    ~Chain();
-    void SetChainName(std::string chainName);
-    void SetChainPolicy(std::string chainPolicy);
-    std::string GetChainName();
-    std::string GetChainPolicy();
-    int GetRuleCount();
-    std::vector<Rule*> GetRules();
-    void Append(Rule* rule);
+    typedef GenericUtility::State State;
 
-private:
-    std::string m_chainName;
-    std::string m_chainPolicy;
-    std::vector<Rule*> m_rules;
+    Iptables() = default;
+    ~Iptables() = default;
+
+    State Detect() const override
+    {
+        // If the utility is not installed/available, the the state is Disabled
+        // If the utility is installed check if there are rules/chain policies in the tables
+        // If there are rules/chain policies, the state is Enabled
+
+        std::string result;
+        return ((0 == utility::Execute("iptables -S", result)) && !result.empty()) ? State::Enabled : State::Disabled;
+    }
+
+    std::string Hash() const override
+    {
+        std::string hash;
+        std::string rules;
+        const std::string command = "iptables -S";
+
+        if (0 == utility::Execute(command.c_str(), rules))
```

Comment:
>utility [](http://example.com/codeflow?start=17&length=7)

Consider renaming this "utility" to something that tells better what this is

Reply:
system_utils:

1. Is not coding style compliant
2. Confuses with commonutils

Why complicate? If an extra wrapper for commonurils!ExecuteComand is needed (is it really needed?) just add a function to current module object, no need for a separate namespace or class.